lib/WTSI/NPG/iRODS/Collection.pm

```diff
@@ -188,19 +190,22 @@ sub get_contents {
    my @collections;
 
    foreach my $obj (@$objs) {
-     my $object = WTSI::NPG::iRODS::DataObject->new($irods, $obj);
+     my ($dobj, $coll) = fileparse($obj);
+     my @init_args = (irods       => $irods,
+                      collection  => $coll,
+                      data_object => $dobj);
 
      if ($checksums) {
        if (exists $object_checksums->{$obj}) {
-         $object->checksum($checksums->{$obj});
+         push @init_args, checksum => $object_checksums->{$obj};
```

Comment:
We are deliberately not validating the value of the checksum that we assign here. The checksum attribute of the WTSI::NPG::iRODS::DataObject class is declared as Str, so assigning an undefined value will give an error. Will be addressed in future if and when this becomes a problem.